clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp

```diff
@@ -8,26 +8,55 @@
 
 #include "QueryAPIMapping.h"
 #include "llvm/Support/raw_ostream.h"
+#include <unordered_map>
 
 namespace clang {
 namespace dpct {
 
-llvm::DenseMap<llvm::StringRef, llvm::StringRef> APIMapping::EntryMap;
+std::unordered_map<std::string, size_t> APIMapping::EntryMap;
+std::vector<llvm::StringRef> APIMapping::EntryArray;
 
-void APIMapping::registerEntry(llvm::StringRef Name,
-                               llvm::StringRef SourceCode) {
-  EntryMap[Name] = SourceCode;
+void APIMapping::registerEntry(std::string Name, llvm::StringRef SourceCode) {
+  std::replace(Name.begin(), Name.end(), '$', ':');
+  // Try to fuzz the original API name:
+  // 1. Remove 0/1/2/... leader '_'.
```

Comment:
```suggestion
  // 1. Remove partial or all leading '_'. 
```

Reply:
for alias, please also consider add fuzzing case:  API_NAME_V1, API_NAME_V2

Reply:
fixed, and no API name has `_v1`, so only add `_v2` fuzz.